khmer/_khmer.cc

```diff
@@ -5215,6 +5215,499 @@ static PyObject * hllcounter_merge(khmer_KHLLCounter_Object * me,
     Py_RETURN_NONE;
 }
 
+/********************************
+ * Assembler classes
+ ********************************/
+
+
+typedef struct {
+    PyObject_HEAD
+    LinearAssembler * assembler;
+} khmer_KLinearAssembler_Object;
+
+#define is_linearassembler_obj(v)  (Py_TYPE(v) == &khmer_KLinearAssembler_Type)
+
+static void khmer_linearassembler_dealloc(khmer_KLinearAssembler_Object * obj)
+{
+    delete obj->assembler;
+    obj->assembler = NULL;
+
+    Py_TYPE(obj)->tp_free((PyObject*)obj);
+}
+
+static PyObject * khmer_linearassembler_new(PyTypeObject *type, PyObject *args,
+                                            PyObject *kwds)
+{
+    khmer_KLinearAssembler_Object *self;
+    self = (khmer_KLinearAssembler_Object*)type->tp_alloc(type, 0);
+
+    if (self != NULL) {
+        PyObject * hashtable_o;
+        Hashtable * hashtable = NULL;
+
+        if (!PyArg_ParseTuple(args, "O", &hashtable_o)) {
+            Py_DECREF(self);
+            return NULL;
+        }
+
+        if (PyObject_TypeCheck(hashtable_o, &khmer_KNodegraph_Type)) {
+            khmer_KHashbits_Object * kho = (khmer_KHashbits_Object *) hashtable_o;
+            hashtable = kho->hashbits;
+        } else if (PyObject_TypeCheck(hashtable_o, &khmer_KCountgraph_Type)) {
+            khmer_KCountingHash_Object * cho = (khmer_KCountingHash_Object *) hashtable_o;
+            hashtable = cho->counting;
+        } else {
+            PyErr_SetString(PyExc_ValueError,
+                            "graph object must be a NodeGraph or CountGraph");
+            Py_DECREF(self);
+            return NULL;
+        }
+
+        try {
+       std::cout << "New Assembler: " << hashtable << std::endl;
```

Comment:
There are a few such debugging print statements outside `#if DEBUG_ASSEMBLY` blocks. Did you intend to leave these in, or should they be removed or marked?